chemportal/cuchemportal/data/db_client.py

```diff
@@ -1,110 +1,78 @@
-        self.session.add_all(entries)
-        self.session.commit()
-        # Todo: Return array of all inserted objects and make a singular version of this method
-        return entries
+        logger.info("Database connected")
+        logger.debug(self.engine.table_names())
 
     def insert(self, record, session: Session):
         """Inserts a pipeline entry  into table"""
-
-        # Adding all entries to session and comitting them
-
+        # Adding record to session
         session.add(record)
-
-        # Todo: Return array of all inserted objects and make a singular version of this method
-        return record
+        session.flush()
+        return record.id
 
     def query_by_id(self, id: int, db_table: Any, session: Session):
         """Obtains all instances in table of each of a list of queries """
 
         # Returning first matching pipeline
         query_result = session.query(db_table).filter(db_table.id == id).first()
-        
+
         return query_result
 
     def query_range(self, db_table: Any, start_idx: int, n_rows: int, session: Session):
         """Obtains first instance in table of each of a list of queries """
          # Returning all pipelines in [start,end)
-        query_result = session.query(db_table).filter((db_table.id >= start_idx)).limit(n_rows)
-        
-        return query_result
+        return session.query(db_table).filter(db_table.is_deleted == 0).offset(start_idx).limit(n_rows).all()
 
-    # TODO: fix update methods
-    def update_record(self, db_table: Any, id: int, new_config: dict, session: Session):
-        """Updates all given database items corresponding to a query""" 
+    def update(self, db_table: Any, record, session: Session):
+        """Updates all given database items corresponding to a query"""
         # Obtaining first value of exact same id (unique so should be only value)
-        updatable = session.query(db_table).filter(db_table.id == id).first()
-        try:
-            for attribute in new_config.keys():
-                # Updating only known attributes on object and mapping back to db
-                if hasattr(updatable, attribute):
-                    setattr(updatable, attribute, new_config[attribute])
-            setattr(updatable, "last_updated", datetime.now())
-        # Preserving atomicity if integrity error found
-        except IntegrityError as e:
-            raise e
-
-        return updatable
+        updatable = session.query(db_table).filter(db_table.id == record.id).first()
+
+        setattr(record, "updated_at", datetime.now())
+        for attribute in updatable.__table__.columns:
+            # Updating only known attributes on object and mapping back to db
+            if hasattr(updatable, attribute.name):
+                setattr(updatable, attribute.name, getattr(record, attribute.name))
+        # setattr(updatable, "updated_at", datetime.now())
+
+        return record
```

Comment:
Don't we want to preserve this method's atomicity in the case of an integrity error? (Allow some sort of rollback)

Reply:
Lets manage transaction in the caller methods. Any failure here will not update database. Since, this is a generic component, we should let the caller decide the transaction boundary.